src/servers/src/tls.rs

```diff
@@ -80,11 +80,21 @@ impl TlsOption {
             .map_err(|_| Error::new(ErrorKind::InvalidInput, "invalid cert"))
             .map(|mut certs| certs.drain(..).map(Certificate).collect())?;
 
-        // TODO(SSebo): support more private key types
-        let key = pkcs8_private_keys(&mut BufReader::new(File::open(&self.key_path)?))
-            .map_err(|_| Error::new(ErrorKind::InvalidInput, "invalid key"))
-            .map(|mut keys| keys.drain(..).map(PrivateKey).next())?
-            .ok_or_else(|| Error::new(ErrorKind::InvalidInput, "invalid key"))?;
+        let key = {
+            let mut pkcs8 = pkcs8_private_keys(&mut BufReader::new(File::open(&self.key_path)?))
```

Comment:
Why not add a configure option to control this logic? Looks like we just try our best to new the private key, it's implicit.

Reply:
Most of the users don't specify the type of private key and the type of private key is **limited**(RSA and PKCS8 in the current version of `rustls/pemfile`). We can automatically parse all the different types of private keys by the above `if...else...` logic. I think this usage will be easier.